cosima_cookbook/database.py

```diff
@@ -330,29 +330,20 @@ def index_experiment(experiment_dir, session=None, client=None, update=False, pr
     update_metadata(expt)
 
     # make all files relative to the experiment path
-    files = [str(Path(f).relative_to(expt_path)) for f in files]
-
-    missing_files = {}
-    if prune:
-        # create a dictionary of all entries that are missing
-        for f in expt.ncfiles:
-            missing_files[f.ncfile] = f
-        for f in files:
-            del(missing_files[f])
-
-    if update:
-        # prune file list to only new files
-        # first construct a query for the current experiment
-        q = (session
-             .query(NCFile.ncfile)
-             .join(NCExperiment)
-             .filter(NCExperiment.experiment == expt.experiment)
-             .filter(NCExperiment.root_dir == expt.root_dir))
-
-        # we can't really leverage the database to do this query, so we'll
-        # just have to iterate over files and see if they're in the database
-        # already. ncfile is an indexed column, so it's not too bad
-        files = [f for f in files if not q.filter(NCFile.ncfile == f).count()]
+    files = set([str(Path(f).relative_to(expt_path)) for f in files])
```

Comment:
Very minor nit: this could be a set comprehension `{str(Path(f) ...}` or `set(Path(f) ...)` rather than converting a list to a set. Doesn't really matter, just noting!

Reply:
Good point. I don't think I've ever seen/made a set comprehension. Thanks for the heads up.